### ISSUE TITLE
Problems detecting words in hiragana

### ISSUE THREAD

**Issue by Wend1go** [birchill/10ten-ja-reader#2105]
I noticed that 10ten has some issues with Hiragana like in this sentence from an NHK headline:
アメリカ　トランプさんがまた大統領になることになった
("nina" - "naruko" instead of "ninaru")
Maybe it would be possible to extend or shrink the selection using the mouse wheel by one letter per turn.

Maybe related to #1969

**Comment by birtles** [comment]
Hi! Thanks for the bug report!

Is the issue that 10ten doesn't recognize になる? If so, I think that's just because the source dictionary (JMdict) doesn't have an entry for になる. It only contains the verb なる without the に particle.

When looking up from な onwards it will show なるこ before なる(成る) because it's longer and it doesn't do any word segmentation, but just looks up from where the cursor is.

Does that make sense or have I misunderstood?

**Comment by Wend1go** [comment]
Yes, this makes sense.
So in order to get 10ten to notice "ninaru", the underlying dictionary needs to be updated.

**Comment by birtles** [comment]
Yes, that's right. However, I don't think JMdict (the source) is in the habit of listing combinations of particles and verbs (since the particle will depend on the preceding context).

**Comment by triadne** [comment]
I came across this issue for the first time today too. 
どうしましたか
is not providing results for どう 
I could sense the sentence structure and understand that dou means how, but i wasn't entirely sure, but the dictionary doesn't recognise it at all. 
I have no memory of this ever occurring before and I've been using this since before it was tenten, maybe more than a decade.

**Comment by birtles** [comment]
That's a tricky one. It's not that the dictionary doesn't recognize どう, it's just that it has so many matches for どうし that it cuts off the results before it gets to どう.

I'm pretty sure it's always been like that, however, and is not a regression.

I think the options here are:

- Make the maximum number of results more generous so you can at least scroll down to the definition for どう
- Submit どう(か)しましたか / どう(か)なさいましたか to JMdict
- Get our deinflection code to deinflect どうしましたか to どうしますか so we can recognize the entry for that
- Do tokenization of the text and prioritize results based on that

**Comment by triadne** [comment]
personally, I'd love to have the option to see more entries if there are many.
The deinflection also sounds incredibly useful.
It's your choice though, of course.

I'm not fluent in Japanese so I'm not always entirely sure how much of what my friend says is slang or improper language, but I do notice his occasional 'spelling' errors from time to time so I guess versatility is most useful.
